### PR TITLE
All kinds of improvements & new features

### PR DESCRIPTION
**Pull request by iPurpl3x**
We (at @glowingblue) would love to use this extension in a productive environment and to do that, we took the time to update this extension's implementation to meet our quality standards.

The main changes in this PR are:
- Use the TextEditor component provided by Flarum to write the details of an event
- Add missing translation keys
- Admin can now optionally hide the event host (default: show)
- Lots of bug-fixes
- Lots of updates to deprecated code
- General code cleanup
- Code formatting (prettier...)
- Dynamically load external libraries, to reduce bundle size
- and much more...

We hope this PR will get accepted and merged into the main repo. Also, if possible, we would love to get special rights on the main repo, to be able to help with maintaining this extension and be able to quickly release any bug-fixes that might be required in the future.

### CODE REVIEW

**Comment by eddiewebb** [comment]
🙏  very grateful and will look this over today!

File: js/dist/forum.js.LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+/*! regenerator-runtime -- Copyright (c) 2014-present, Facebook, Inc. -- license (MIT): https://github.com/facebook/regenerator/blob/main/LICENSE */
```
Comment (eddiewebb) [inline js/dist/forum.js.LICENSE.txt:1]: Not sure about this..

File: js/src/forum/utils/dynamicallyLoadLib.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
+const libs = {
```
Comment (eddiewebb) [inline js/src/forum/utils/dynamicallyLoadLib.js:1]: Brilliant! shrinks size WAYYYYY down!

**Review by eddiewebb — COMMENTED** [review]
Code changes look good. I've been lacking on any CI for this repo so will pull down for local validation tonight.

**Review by eddiewebb — CHANGES_REQUESTED** [review]
Qualify FB generator usage if required to keep license here.

**Review by eddiewebb — APPROVED** [review]
local testing looks good will merge and tag.

**Comment by eddiewebb** [comment]
https://packagist.org/packages/webbinaro/flarum-calendar as 1.6.0

**Comment by iPurpl3x** [comment]
Cool, thanks for merging and releasing my changes @eddiewebb !

**Pull request merged**